frontend/src/lib/components/DefinitionPopup/definitionPopupLogic.ts

```diff
@@ -1,21 +1,43 @@
 import { kea } from 'kea'
+import { router } from 'kea-router'
 import { definitionPopupLogicType } from './definitionPopupLogicType'
+import { TaxonomicDefinitionTypes, TaxonomicFilterGroupType } from 'lib/components/TaxonomicFilter/types'
+import { capitalizeFirstLetter, errorToast } from 'lib/utils'
+import { getSingularType } from 'lib/components/DefinitionPopup/utils'
+import { ActionType, CohortType, EventDefinition, PropertyDefinition } from '~/types'
+import { urls } from 'scenes/urls'
+import api from 'lib/api'
+import { eventDefinitionsModel } from '~/models/eventDefinitionsModel'
+import { actionsModel } from '~/models/actionsModel'
+import { propertyDefinitionsModel } from '~/models/propertyDefinitionsModel'
+import { cohortsModel } from '~/models/cohortsModel'
+import { toast } from 'react-toastify'
+import equal from 'fast-deep-equal'
 
-import { TaxonomicFilterGroupType } from 'lib/components/TaxonomicFilter/types'
 export enum DefinitionPopupState {
     Edit = 'edit',
     View = 'view',
 }
 
-interface DefinitionPopupLogicProps {
-    type: TaxonomicFilterGroupType
+export interface DefinitionPopupLogicProps {
+    /* String type accounts for types with `TaxonomicFilterGroupType.GroupsPrefix` prefix */
+    type: TaxonomicFilterGroupType | string
+    hasTaxonomyFeatures?: boolean
+    /* Callback to update specific item in in-memory list */
+    updateRemoteItem?: (item: Partial<TaxonomicDefinitionTypes>) => void
 }
 
 export const definitionPopupLogic = kea<definitionPopupLogicType<DefinitionPopupLogicProps, DefinitionPopupState>>({
     props: {} as DefinitionPopupLogicProps,
     path: ['lib', 'components', 'DefinitionPanel', 'definitionPopupLogic'],
     actions: {
+        setDefinition: (item: Partial<TaxonomicDefinitionTypes>) => ({ item }),
+        setLocalDefinition: (item: Partial<TaxonomicDefinitionTypes>) => ({ item }),
         setPopupState: (state: DefinitionPopupState) => ({ state }),
+        setNewTag: (tag: string) => ({ tag }),
+        deleteTag: (tag: string) => ({ tag }),
```

Comment:
Note this api will change once https://github.com/PostHog/posthog/pull/8567 is merged in. This PR will probably be merged in after that one.